src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -67,16 +72,104 @@ internal struct ParsedStat
             //internal long cguest_time;
         }
 
+        internal struct ParsedMapsModule
+        {
+            internal string FileName;
+            internal KeyValuePair<long, long> AddressRange;
+        }
+
         internal static string GetStatFilePathForProcess(int pid)
         {
             return RootPath + pid.ToString(CultureInfo.InvariantCulture) + StatFileName;
         }
 
+        internal static string GetMapsFilePathForProcess(int pid)
+        {
+            return RootPath + pid.ToString(CultureInfo.InvariantCulture) + MapsFileName;
+        }
+
         internal static string GetTaskDirectoryPathForProcess(int pid)
         {
             return RootPath + pid.ToString(CultureInfo.InvariantCulture) + TaskDirectoryName;
         }
 
+        internal static IEnumerable<ParsedMapsModule> ParseMapsModules(int pid)
+        {
+            try
+            {
+                return ParseMapsModulesCore(File.ReadAllLines(GetMapsFilePathForProcess(pid)));
+            }
+            catch (FileNotFoundException) { }
+            catch (DirectoryNotFoundException) { }
+            catch (UnauthorizedAccessException) { }
+
+            return Array.Empty<ParsedMapsModule>();
+        }
+
+        private static IEnumerable<ParsedMapsModule> ParseMapsModulesCore(IEnumerable<string> lines)
+        {
+            Debug.Assert(lines != null);
+
+            // Parse each line from the maps file into a ParsedMapsModule result
+            foreach (string line in lines)
+            {
+                // Use a StringParser to avoid string.Split costs
+                var parser = new StringParser(line, separator: ' ', skipEmpty: true);
+
+                // Parse the address range 
+                KeyValuePair<long, long> addressRange =  
+                    parser.ParseRaw(delegate (string s, ref int start, ref int end)
```

Comment:
Should this be codified in the Coding Style, or is it not worth the extra rule for such a trivial thing?

The reason I asked is because the MSDN [documentation](https://msdn.microsoft.com/en-us/library/0yw3tz5k.aspx) prefers the use of the lambda syntax over anonymous method syntax:

> C# 2.0 introduced anonymous methods and in C# 3.0 and later, lambda expressions supersede anonymous methods as the preferred way to write inline code.

Unless omitting the parameter list:

> There is one case in which an anonymous method provides functionality not found in lambda expressions. Anonymous methods enable you to omit the parameter list. This means that an anonymous method can be converted to delegates with a variety of signatures. This is not possible with lambda expressions.

I realize that when you have to specify the parameter types (because of ref), there isn't much difference between `delegate (string s, ref int start, ref int end) { ... }` and `(string s, ref int start, ref int end) => { ... }`.

I was just wondering if the rule should be "Prefer the lambda syntax, unless there is a need to omit the parameter list, in which case the anonymous method syntax can be used".